File: Domain/Company.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Domain
{
    public class Company
    {
        public Data Data {get; set;}
    }

    public class Data
    {
        public string DateListed {get; set;}

        public string DisplayName {get; set;}

        public double PriceAsk { get; set; }

        public double PriceBid { get; set; }

         public double PriceLast { get; set; }

        public string Sector {get; set;}

         public string Symbol { get; set; }
    }
}
// {
//     "data": {
//         "dateListed": "1885-08-13",
//         "displayName": "BHP GROUP LIMITED",
//         "priceAsk": 38.29,
//         "priceBid": 38.25,
//         "priceChange": 0.9200000000000017,
//         "priceChangePercent": 2.4631860776439134,
//         "priceLast": 38.27,
//         "sector": "Materials",
//         "industryGroup": "Materials",
//         "securityType": 1,
//         "symbol": "BHP",
//         "volume": 8163812,
//         "xid": "60947",
//         "marketCap": 189077771147,
//         "statusCode": ""
//     }
// }

File: next-app/__mocks__/fakeData/Company.ts
import { Company } from "../../models/company";

export const fakeCompany:Company = { 
    displayName: "COMPUTERSHARE LIMITED.",
    symbol: "CPU",
    dateListed: "1994-05-27",
    sector: "Industrials",
    priceAsk: 22.7,
    priceBid: 22.150000000000002, 
    priceLast: 22.400000000000002
};

export const badCompany:Company = { 
    displayName: "BAD CO LIMITED.",
    symbol: "CPX",
    dateListed: "1991-05-27",
    sector: "None",
    priceAsk:12.7,
    priceBid: 12.150000000000002, 
    priceLast: 12.400000000000002
};

